### ISSUE TITLE
Rule TRY302 raised while the re-raise changes the error

### ISSUE THREAD

**Issue by 153957** [charliermarsh/ruff#4548]
With code like the example below I got this unexpected error:
> TRY302 Remove exception handler; error is immediately re-raised

Although the error is immediately re-raised, it is modified, the causes are removed.

Code to reproduce:
```python
try:
    self.assertEqual(cells, interlinked)
except AssertionError as error:
    # Catch and raise again without causes
    raise error from None
```

Enabled ALL settings.
Version: ruff 0.0.269

**Comment by JonathanPlasse** [comment]
Good catch!

**Comment by JonathanPlasse** [comment]
Would you like to work on it?

**Comment by 153957** [comment]
I'd like to, but have no experience writing Rust, so probably can't without some pointers.

I assume fixing this requires changes to the logic here: https://github.com/charliermarsh/ruff/blob/fc63c6f2e260760963c5fdea2d9d4e0cf38ec58a/crates/ruff/src/rules/tryceratops/rules/useless_try_except.rs#L52-L55
Somehow checking it there is a `from` after the raise statement, i.e. following the expression for which the name is checked, if that is the case just `return None;`.

And add a new test case(s) here: https://github.com/charliermarsh/ruff/blob/fc63c6f2e260760963c5fdea2d9d4e0cf38ec58a/crates/ruff/resources/test/fixtures/tryceratops/TRY302.py#L4

**Comment by JonathanPlasse** [comment]
You can look up `cause` in `ast::StmtRaise`.

**Comment by 153957** [comment]
Thank you for the hint, PR is up.
